test/json_unittest.cc

```diff
@@ -0,0 +1,385 @@
+#include <functional>
+
+#include "../src/json.h"
+#include "gtest/gtest.h"
+
+#define EXPECT_TO_STRING(v, s) EXPECT_EQ(v->ToString(), s)
+
+namespace {
+
+void GuardJsonException(std::function<void()> test) {
+  try {
+    test();
+  } catch (const json::Exception& exc) {
+    FAIL() << exc.what();
+  }
+}
+
+
+// Trival Null
+
+TEST(TrivialParseTest, Null) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("null");
+    EXPECT_EQ(v->type(), json::NullType);
+  });
+}
+
+TEST(TrivialToStringTest, Null) {
+  GuardJsonException([](){
+    EXPECT_TO_STRING(json::null(), "null");
+  });
+}
+
+
+// Trival True & False
+
+TEST(TrivialParseTest, True) {
+  GuardJsonException([](){
+    std::unique_ptr<json::Value> v = json::Parser::FromString("true");
```

Comment:
`std::unique_ptr<json::Value>` is `json::value`. Let's just use that.

Reply:
Done.